### PR TITLE
Feat: M2-2652 - Builder: save and publish data, received from the library

### PR DESCRIPTION
**Pull request by muhurov**

### CODE REVIEW

File: src/modules/Library/features/Popups/AddToBuilderPopup/AddToBuilderPopup.utils.tsx [hunk 1]
```diff
@@ -282,9 +296,41 @@ export const getPerformanceTaskType = (responseType: ItemResponseType) => {
   return performanceTaskType;
 };
 
-// TODO: Make filtering for scores, subscales, and item flows, regarding which items were selected, Activity to the
-//  Reviewer dashboard assessment checkbox (can be only one in all activities), names for ABTrails activities to be
-//  unique
+const getUniqueActivityName = <T extends { name: string }>(
```
Comment (IhorNesterenko99) [inline src/modules/Library/features/Popups/AddToBuilderPopup/AddToBuilderPopup.utils.tsx:299]: it seems the same function is in builder, pls avoid code duplication

File: src/modules/Library/features/Popups/AddToBuilderPopup/AddToBuilderPopup.utils.tsx [hunk 2]
```diff
@@ -282,9 +296,41 @@ export const getPerformanceTaskType = (responseType: ItemResponseType) => {
...
+const mapResponseValues = <
+  T extends {
+    dataMatrix?: SingleAndMultipleSelectMatrix[] | null;
+    options: (SingleAndMultiSelectOption | SingleAndMultiSelectRowOption)[];
+  },
+>(
+  responseValues: T,
+): T => ({
+  ...responseValues,
+  ...(!responseValues.dataMatrix && {
```
Comment (IhorNesterenko99) [inline src/modules/Library/features/Popups/AddToBuilderPopup/AddToBuilderPopup.utils.tsx:320]: maybe 
options: !!responseValues.dataMatrix ? ... : ...

File: src/modules/Library/features/Popups/AddToBuilderPopup/AddToBuilderPopup.utils.tsx [hunk 3]
```diff
@@ -309,21 +379,29 @@ export const getSelectedAppletData = (
...
 
+      //per requirements if not all of the items in activity were selected, scores & reports and subscales should be removed
+      const hasSubscalesAndScores = filteredItems.length === activity.items.length;
+
       return {
         ...activity,
         isPerformanceTask,
         performanceTaskType: performanceTaskType || undefined,
         items: filteredItems,
+        subscaleSetting: hasSubscalesAndScores ? activity.subscaleSetting : undefined,
```
Comment (IhorNesterenko99) [inline src/modules/Library/features/Popups/AddToBuilderPopup/AddToBuilderPopup.utils.tsx:390]:
```suggestion
        ...(hasSubscalesAndScores && { subscaleSetting: activity.subscaleSetting }),
```

File: src/modules/Library/features/Popups/AddToBuilderPopup/AddToBuilderPopup.utils.tsx (hunk 1 above)
Comment (muhurov) [reply src/modules/Library/features/Popups/AddToBuilderPopup/AddToBuilderPopup.utils.tsx]: done

File: src/modules/Library/features/Popups/AddToBuilderPopup/AddToBuilderPopup.utils.tsx (hunk 2 above)
Comment (muhurov) [reply src/modules/Library/features/Popups/AddToBuilderPopup/AddToBuilderPopup.utils.tsx]: done

File: src/modules/Library/features/Popups/AddToBuilderPopup/AddToBuilderPopup.utils.tsx (hunk 3 above)
Comment (muhurov) [reply src/modules/Library/features/Popups/AddToBuilderPopup/AddToBuilderPopup.utils.tsx]: done

File: src/modules/Library/features/Popups/AddToBuilderPopup/AddToBuilderPopup.utils.tsx [hunk 4]
```diff
@@ -300,7 +346,31 @@ export const getSelectedAppletData = (
...
         .filter((item) => selectedItemNamesSet.has(`${item.name}-${activity.name}`))
-        .map((item) => ({ ...item, key: uuidv4() }));
+        .map((item, index, items) => {
+          const newItem = {
+            ...item,
+            key: uuidv4(),
+          };
+
+          //for security reasons there is no 'id' in responseValues.options for Single/Multi selection
+          if (responseTypeToHaveOptions.includes(newItem.responseType)) {
```
[the remark this answers is not in the document]
Comment (muhurov) [reply src/modules/Library/features/Popups/AddToBuilderPopup/AddToBuilderPopup.utils.tsx:342]: done

File: src/modules/Builder/features/Activities/Activities.tsx [hunk 5]
```diff
@@ -105,7 +97,7 @@ export const Activities = () => {
         ? performanceTaskName
         : t('newActivity');
 
-    const name = getAddedActivityName(newActivityName, activityNames, setActivityNames);
+    const name = getUniqueName(newActivityName, pluck(activities, 'name'));
```
Comment (IhorNesterenko99) [inline src/modules/Builder/features/Activities/Activities.tsx:100]: maybe use separate variable for pluck(activities, 'name')?

Comment (muhurov) [reply src/modules/Builder/features/Activities/Activities.tsx]: done

File: src/modules/Library/features/Popups/AddToBuilderPopup/AddToBuilderPopup.utils.tsx [hunk 6]
```diff
@@ -282,9 +293,30 @@ export const getPerformanceTaskType = (responseType: ItemResponseType) => {
...
+const mapResponseValues = <
+  T extends {
+    dataMatrix?: SingleAndMultipleSelectMatrix[] | null;
+    options: (SingleAndMultiSelectOption | SingleAndMultiSelectRowOption)[];
+  },
+>(
+  responseValues: T,
+): T => ({
+  ...responseValues,
+  ...(!responseValues.dataMatrix
```
Comment (IhorNesterenko99) [inline src/modules/Library/features/Popups/AddToBuilderPopup/AddToBuilderPopup.utils.tsx:305]: pls make direct ternary:
responseValues.dataMatrix ? ... : ...

**Review by IhorNesterenko99 — APPROVED** [review]

File: src/modules/Library/features/Popups/AddToBuilderPopup/AddToBuilderPopup.utils.tsx (hunk 6 above)
Comment (muhurov) [reply src/modules/Library/features/Popups/AddToBuilderPopup/AddToBuilderPopup.utils.tsx]: done

**Review by kdanilenka32 — APPROVED** [review]

**Pull request merged**